runtime/src/main/scala/akka/grpc/GrpcClientResponse.scala

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.grpc
+
+import java.util.Optional
+import java.util.concurrent.CompletionStage
+
+import akka.annotation.DoNotInherit
+import io.grpc.Metadata
+
+import scala.concurrent.Future
+
+// FIXME should we provide our own immutable/thread safe response metadata abstraction?
```

Comment:
Indeed the upstream `Metadata` is not too nice, might be good to at least wrap it in some immutable interface?.

Reply:
Done!